voice-interface/src/main/java/com/derpgroup/derpwizard/voice/util/ConversationHistoryUtils.java

```diff
@@ -0,0 +1,46 @@
+package com.derpgroup.derpwizard.voice.util;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import com.derpgroup.derpwizard.voice.model.CommonMetadata;
+import com.derpgroup.derpwizard.voice.model.ConversationHistoryEntry;
+import com.derpgroup.derpwizard.voice.model.VoiceInput;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class ConversationHistoryUtils {
+
+  public static ConversationHistoryEntry getLastNonMetaRequestBySubject(Deque<ConversationHistoryEntry> conversationHistory, Set<String> metaSubjects){ //Lets talk about what this should be named
+    
+    for(ConversationHistoryEntry entry : conversationHistory){
+      if(entry.getMessageSubject() != null && !metaSubjects.contains(entry.getMessageSubject())){
+        return entry;
+      }
+    }
+    
+    return null; //Should this return null?
+  }
+  
+  //Should this operate on the object directly, or return something?
+  public static void appendToConversationHistory(@NonNull VoiceInput voiceInput, CommonMetadata metadata, ObjectMapper mapper) {
```

Comment:
This is the other side of my other comment. CommonMetadata is an internal variable inside VoiceInput. So why are we passing both here?